### PR TITLE
beagle 4.0.0

### PR DESCRIPTION
**Pull request by thanasibakis**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

New version of [beagle](https://github.com/beagle-dev/beagle-lib/releases/tag/v4.0.0). The developers modified the build process since the last version, so the formula needed adjustments in `install` and `depends_on`.

### CODE REVIEW

File: Formula/beagle.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Beagle < Formula
   desc "Evaluate the likelihood of sequence evolution on trees"
   homepage "https://github.com/beagle-dev/beagle-lib"
-  url "https://github.com/beagle-dev/beagle-lib/archive/v3.1.2.tar.gz"
-  sha256 "dd872b484a3a9f0bce369465e60ccf4e4c0cd7bd5ce41499415366019f236275"
-  license "LGPL-3.0-or-later"
+  url "https://github.com/beagle-dev/beagle-lib/archive/v4.0.0.tar.gz"
+  sha256 "d197eeb7fe5879dfbae789c459bcc901cb04d52c9cf5ef14fb07ff7a6b74560b"
+  license "MIT"
   revision 1
```
Comment (carlocab) [inline Formula/beagle.rb:7]:
```suggestion
```
We remove the `revision` on version bumps.

File: Formula/beagle.rb [hunk 2]
```diff
@@ -22,19 +22,20 @@ class Beagle < Formula
...
+    mkdir "build" do
+      system "cmake",
+        "-DJAVA_HOME=#{Formula["openjdk"].opt_prefix}",
+        "-DJAVA_AWT_LIBRARY=#{Formula["openjdk"].include}",
+        "-DJAVA_JVM_LIBRARY=#{Formula["openjdk"].include}",
+        "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}",
+        ".."
+      system "make"
+      system "make", "install"
+    end
```
Comment (carlocab) [inline Formula/beagle.rb:38]: Let's follow the pattern used by `brew create --cmake`.
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DJAVA_HOME=#{Formula["openjdk"].opt_prefix}",
                    "-DJAVA_AWT_LIBRARY=#{Formula["openjdk"].include}",
                    "-DJAVA_JVM_LIBRARY=#{Formula["openjdk"].include}",
                    *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

**Review by carlocab — COMMENTED** [review]
Thanks, @thanasibakis. A few suggestions.

**Comment by thanasibakis** [comment]
These sound great, thanks for the feedback!

**Comment by carlocab** [comment]
Looks like `libhmsbeagle` is no longer versioned. Previously it was `libhmsbeagle.1.dylib`. Now, it is `libhmsbeagle.dylib`, and that's broken `mrbayes`. Would you know if this is intentional?

This issue is actually pretty typical of a build system change, so I suspect that it is not.

**Comment by thanasibakis** [comment]
I'm actually not sure at all, unfortunately. For what it might be worth, I'm using beagle in conjunction with the [beast formula](https://formulae.brew.sh/formula/beast#default), which marks beagle as a dependency, and haven't seen this kind of issue.

**Comment by SMillerDev** [comment]
Could you ask upstream if this is intentional?

**Comment by chenrui333** [comment]
#111894

**Pull request closed without merging**